src/internal/packager/git/checkout.go

```diff
@@ -21,28 +21,42 @@ func (g *Git) CheckoutTag(tag string) {
 	g.checkout(options)
 }
 
+func (g *Git) checkoutRefAsBranch(ref string, branch plumbing.ReferenceName) error {
+	var err error
+
+	if IsHash(ref) {
+		err = g.checkoutHashAsBranch(plumbing.NewHash(ref), branch)
+	} else {
+		err = g.checkoutTagAsBranch(ref, branch)
+	}
+
+	return err
+}
+
 // checkoutTagAsBranch performs a `git checkout` of the provided tag but rather
 // than checking out to a detached head, checks out to the provided branch ref
 // It will delete the branch provided if it exists.
-func (g *Git) checkoutTagAsBranch(tag string, branch plumbing.ReferenceName) {
+func (g *Git) checkoutTagAsBranch(tag string, branch plumbing.ReferenceName) error {
 	message.Debugf("git.checkoutTagAsBranch(%s,%s)", tag, branch.String())
 
 	repo, err := git.PlainOpen(g.GitPath)
 	if err != nil {
-		message.Fatal(err, "Not a valid git repo or unable to open")
+		message.Warnf("Not a valid git repo or unable to open")
```

Comment:
Should be using `message.Warf()` here or just return a formatted error like we do most other places in the codebase?